tap-version-14-specification.md

```diff
@@ -0,0 +1,741 @@
+---
+layout: default
+title: TAP 14 specification
+---
+
+# NAME
+
+TAP14 - The Test Anything Protocol v14
+
+## SYNOPSIS
+
+TAP, the Test Anything Protocol, is a simple text-based interface
+between testing modules a test harness. TAP started life as part of
+the test harness for Perl but now has implementations in C/C++,
+Python, PHP, Perl, JavaScript, MatLab, and probably others by the time
+you read this.
+
+This document describes version 14 of TAP.
+
+## THE TAP14 FORMAT
+
+A TAP stream is a UTF-8 encoded text stream representing a set of
+tests.  It should be parsed line by line.  Lines are separated by
+`\n`.
+
+TAP14's general format is:
+
+```
+TAPDocument ->
+  Version Plan Body |
+  Version Body Plan
+```
+
+The Version is always `TAP version 14`.
+
+The Body is a collection of lines representing a test set.
+
+The Plan reports on the number of tests included in the Body.
+
+Any line matching the pattern `/^\s*#.*$/` is treated as a diagnostic
+comment, and MAY be reported to the user or ignored, but MUST be
+ignored for the purposes of interpreting test output semantics.
```

Comment:
`/^\s*#.*$/` always being ignored as a comment sounds great, but only works with JSON diagnostics, not YAML diagnostics. The [YAML block literal](https://en.wikipedia.org/wiki/YAML#Block_literals) can break this:

```yaml
data: >
    This is YAML data and it might just contain a hash character (
    #) and have this line ignored because TAP 14 thinks it's a
    comment.
```